Probable typo for IPv6 size in MacOS

When compiling on MacOS Catalina:
```
mdns/mdns.h:646:27: error: use of undeclared identifier 'saddr6'
                addr6.sin6_len = sizeof(saddr6);
```

It looks like 
https://github.com/mjansson/mdns/blob/7e1050d26bf9808ad25a841739ab17b7085d08a3/mdns.h#L646
should be
```
addr6.sin6_len = sizeof(addr6);
```

--- a/mdns.h
+++ b/mdns.h
@@ -643,7 +643,7 @@ mdns_multicast_send(int sock, const void* buffer, size_t size) {
 		memset(&addr6, 0, sizeof(addr6));
 		addr6.sin6_family = AF_INET6;
 #ifdef __APPLE__
-		addr6.sin6_len = sizeof(saddr6);
+		addr6.sin6_len = sizeof(addr6);
 #endif
 		addr6.sin6_addr.s6_addr[0] = 0xFF;
 		addr6.sin6_addr.s6_addr[1] = 0x02;

